apps/web/modules/bookings/views/bookings-single-view.tsx

```diff
@@ -122,6 +122,13 @@ export default function Success(props: PageProps) {
   } = querySchema.parse(routerQuery);
   const attendeeTimeZone = bookingInfo?.attendees.find((attendee) => attendee.email === email)?.timeZone;
 
+  //check if booking has not already started
+  const hasNotStarted = () => {
```

Comment:
First, it would be good to wrap `hasNotStarted` with `useMemo` to run the app more efficiently.

Next, I am thinking that rather than not allowing to reschedule if the booking has started, hide the "reschedule" button if booking is in the past:
```
const isBookingInPast = new Date(booking.endTime) < new Date();
```

My rationale: if Bob books me at 10AM and is late, and realizes that he can't actually make it to the meeting, he can go to the bookings page and re-schedule it. The same goes for your changes in the [BookingListItem.tsx](https://github.com/calcom/cal.com/pull/15304/files#diff-178e89024734f552c5fa0953359ded5880f0e6975ae88e7bb085b3908c762abd) - if Bob is late, I could offer him that I re-schedule the meeting to another time, but [BookingListItem.tsx](https://github.com/calcom/cal.com/pull/15304/files#diff-178e89024734f552c5fa0953359ded5880f0e6975ae88e7bb085b3908c762abd) currently hides re-scheduling.

I know Shiraz wrote in the linked issue #14742 that "reschedule" should not appear before start time, but I think it makes more sense to not show it only after the meeting time has ended, because the root problem, as I see it, are re-schedules that come days or even weeks after the event ended.

I think that reverting changes in [BookingListItem.tsx](https://github.com/calcom/cal.com/pull/15304/files#diff-178e89024734f552c5fa0953359ded5880f0e6975ae88e7bb085b3908c762abd) and changing logic in  [bookings-single-view.tsx](https://github.com/calcom/cal.com/pull/15304/files#diff-a8167bc98af719ed1c40f876ba4bfdd45df5a4df80098ed133b7695211991504) to `isBookingInPast` gives bit more flexibility while dis-allowing people re-scheduling bookings days after the date.

Finally, In [BookingListItem.tsx](https://github.com/calcom/cal.com/pull/15304/files#diff-178e89024734f552c5fa0953359ded5880f0e6975ae88e7bb085b3908c762abd) we have "upcoming" tab in UI, but having introduced `edit_booking_upcoming` and `edit_booking_started` would also require adding "In progress" section, because it's not clear to user why for some bookings 2 options and for others 3 options can be picked under edit. That's why I would just leave it, because also it has not been reported as a problem to solve in the first place.

What do you think?